web3/utils/validation.py

```diff
@@ -18,6 +38,59 @@ def validate_abi(abi):
             raise ValueError("The elements of 'abi' are not all dictionaries")
 
 
+def validate_abi_type(abi_type):
+    """
+    Helper function for validating an abi_type
+    """
+    if not is_recognized_type(abi_type):
+        raise ValueError("Unrecognized abi_type: {abi_type}".format(abi_type=abi_type))
+
+
+def validate_abi_value(abi_type, value):
+    """
+    Helper function for validating a value against the expected abi_type
+    Note: abi_type 'bytes' must either be python3 'bytes' object or '0x'
+    prefixed string
+    """
+    correct = True
```

Comment:
I'd like to see the mutability removed from this function and instead have `return` statements inside the `if/elif` blocks.  Then, if the function has not exited by the point where the `if/elif/else` block finishes you can raise the exception there or in the final `else` statement.